### PR TITLE
Deprecation: remove code for config file v1 and default config file

### PR DESCRIPTION
**Pull request by humitos**
This is a continuation of the work done for

* https://github.com/readthedocs/readthedocs.org/issues/10351
* https://github.com/readthedocs/readthedocs.org/issues/10342
* https://github.com/readthedocs/readthedocs.org/issues/10348
* https://github.com/readthedocs/readthedocs.org/pull/10355

In that work we added notifications, warning messages and finally a feature flag to start failing the builds without a configuration file or using v1.

This commit removes all the code for config file v1 and building without a configuration file. Due to that, it also removes the code that allows users to use the older Docker images (testing, latest, stable)

This work can be merged once we enable the feature flag to fail the builds for all the projects and we are happy with the results.

Related:
* https://github.com/readthedocs/readthedocs.org/pull/10589

### CODE REVIEW

**Comment by humitos** [comment]
> This commit removes all the code for config file v1 and building without a configuration file. Due to that, it also removes the code that allows users to use the older Docker images (testing, latest, stable)

I created a Metabase question to know how many projects are still manually specifying `build.image` on their config file: https://ethicalads.metabaseapp.com/question/405-projects-specifying-build-image-config-file?days=90. In the last 90 days, we currently have **~1.1k projects using `build.image` on their config file**

**Comment by humitos** [comment]
Update this PR to `disable` the "Advanced settings" from Admin page: https://github.com/readthedocs/ext-theme/issues/159#issuecomment-1699595905

**Comment by humitos** [comment]
> This commit removes all the code for config file v1 and building without a configuration file. Due to that, it also removes the code that allows users to use the older Docker images (testing, latest, stable)

We are deprecating this on October 16th (see https://github.com/readthedocs/readthedocs.org/issues/10587), so we may want to merge this _after_ that day or adapt this PR to keep `build.image` for now.

**Comment by ericholscher** [comment]
@humitos I saw merging this in the release tasks, but seems it's not ready?

**Comment by humitos** [comment]
He, yeah. I added this PR into this release a long time ago expecting to be ready by this time 😅

I'll update it in the next few days and ask for a review.

**Comment by humitos** [comment]
Due to https://github.com/readthedocs/readthedocs.org/pull/10367#issuecomment-1705328681, I think we can merge this PR _after_ October 16th and reduce the workload in our side.

**Comment by humitos** [comment]
This PR is ready to review and merge. After that, we can also merge https://github.com/readthedocs/readthedocs-ops/pull/1383 to remove old Docker images from AMI and builders.

**Comment by humitos** [comment]
@stsewd @ericholscher this PR removes everything related to config file v1 and default config file. I'd appreciate a review on it so we can move forward with this deprecation and finally remove this code from our repository.

Note that there is no need to do an extensive review on it since it's pretty big and, in theory, there is no changes in logic or similar. All the code removed from this PR should not be used by anybody at this point since we already passed the final removal date.

Also, since it changes 5k lines, I'd appreciate if we can merge it sooner and make the needed/small changes in another PR if required to avoid having to deal with merge conflicts again.

**Review by stsewd — APPROVED** [review]

File: docs/user/config-file/v1.rst
[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/config-file/v1.rst:1]: Good call! Done 👍🏼

File: readthedocs/doc_builder/python_environments.py [hunk 1]
```diff
@@ -204,80 +199,6 @@ def _install_latest_requirements(self, pip_install_cmd):
             cwd=self.checkout_path,
         )
 
-    def _install_old_requirements(self, pip_install_cmd):
```
Comment (stsewd) [inline readthedocs/doc_builder/python_environments.py:207]: Looks like we are also missing removing these for buils using conda

File: readthedocs/projects/models.py [hunk 2]
```diff
@@ -587,6 +518,100 @@ class Project(models.Model):
         blank=True,
     )
 
+    # TODO: remove the following fields since they all are going to be ignored
```
Comment (ericholscher) [inline readthedocs/projects/models.py:521]: ❤️

File: readthedocs/settings/base.py [hunk 3]
```diff
@@ -553,20 +553,6 @@ def TEMPLATES(self):
...
-        #     'task': 'readthedocs.projects.tasks.utils.deprecated_config_file_used_notification',
-        #     'schedule': crontab(day_of_week='wed', hour=11, minute=15),
-        #     'options': {'queue': 'web'},
-        # },
-        # 'weekly-build-image-notification': {
-        #     'task': 'readthedocs.projects.tasks.utils.deprecated_build_image_notification',
-        #     'schedule': crontab(day_of_week='wed', hour=9, minute=15),
-        #     'options': {'queue': 'web'},
-        # },
     }
```
[the remark this answers is not in the document]
Comment (humitos) [reply readthedocs/settings/base.py:556]: I don't think so. The deprecation time is already over. Both of them are already deprecated.

**Pull request merged**